nnfabrik/templates/checkpoint.py

```diff
@@ -0,0 +1,166 @@
+import copy
+from typing import Dict, Tuple
+
+import datajoint as dj
+import tempfile
+import torch
+import os
+from nnfabrik.main import Model, Dataset, Trainer, Seed, Fabrikant
+from nnfabrik.builder import get_all_parts, get_model, get_trainer
+from nnfabrik.templates.trained_model import TrainedModelBase
+from nnfabrik.utility.dj_helpers import make_hash, clone_conn, CustomSchema
+from nnfabrik.builder import resolve_data
+from datajoint.fetch import DataJointError
+from nnfabrik.main import schema
+
+
+def my_checkpoint(nnfabrik):
+    conn_clone = clone_conn(dj.conn())
+    schema_clone = CustomSchema(nnfabrik.schema.database, connection=conn_clone)
+
+    @schema_clone
+    class Checkpoint(dj.Manual):
+        storage = "minio"
+
+        @property
+        def definition(self):
+            definition = """
+            # Checkpoint table
+            -> nnfabrik.Trainer
+            -> nnfabrik.Dataset
+            -> nnfabrik.Model
+            -> nnfabrik.Seed
+            epoch:                             int          # epoch of creation
+            ---
+            score:                             float        # current score at epoch
+            state:                             attach@{storage}  # current state
+            ->[nullable] nnfabrik.Fabrikant
+            trainedmodel_ts=CURRENT_TIMESTAMP: timestamp    # UTZ timestamp at time of insertion
+            """.format(
+                storage=self.storage
+            )
+            return definition
+
+    return Checkpoint
+
+
+class TrainedModelChkptBase(TrainedModelBase):
+    checkpoint_table = None  # Checkpoint
+    keys = [
+        "model_fn",
+        "model_hash",
+        "dataset_fn",
+        "dataset_hash",
+        "trainer_fn",
+        "trainer_hash",
+    ]
+
+    def call_back(self, uid=None, epoch=None, model=None, state=None):
```

Comment:
It would be nice to have a way to save every nth epoch instead of every single one.